geopyspark/geotrellis/geotiff_rdd.py

```diff
@@ -1,37 +1,114 @@
+                    then the default value is: 'yyyy:MM:dd HH:mm:ss".
+                maxTileSize (int, optional): The max size of each tile in the
+                    resulting RDD. If the size is smaller than a read in tile,
+                    then that tile will be broken into tiles of the specified
+                    size. If None, then the whole tile will be read in.
+                numPartitions (int, optional): The number of repartitions Spark
+                    will make when the data is repartitioned. If None, then the
+                    data will not be repartitioned.
+                chunkSize (int, optional): How many bytes of the file should be
+                    read in at a time. If None, then files will be read in 65536
+                    byte chunks.
+
+            S3 has the above options in addition to this:
+                s3Client (st, optional): Which S3Cleint to use when reading
+                    GeoTiffs. There are currently two options: 'default' and
+                    'mock'. If None, 'defualt' is used. Note: 'mock' should
+                    only be used in unit tests.
+
+        **kwargs: Option parameters can also be entered as keyword arguements.
+            Note: Defining both options and keyword arguements will cause the
+            keyword arguements to be ignored in favor of options.
 
-        ser = self.geopysc.create_tuple_serializer(result._2(), value_type="Tile")
+    Returns:
+        RDD: A RDD that contains tuples of dictionaries, (K, V).
+            K (dict): The projected extent information of the GeoTiff.
+                Which is the pyhiscal extent and the CRS of the GeoTiff.
+            V (dict): The raster tile information of the GeoTiff.
+                Which is the Tile data itself represented as a numpy array,
+                and the no_data_value of the Tile, if any.
 
-        return self.geopysc.create_python_rdd(result._1(), ser)
+    Examples:
+        Reading a local GeoTiff that does not have a time component with the
+        default options.
 
+        >>> geotiff_rdd(geopysc, SPATIAL, 'file://path/to/my/geotiff.tif')
+        RDD
 
-class S3GeoTiffRDD(object):
-    def __init__(self, geopysc):
-        self.geopysc = geopysc
-        self._s3_wrapper = self.geopysc.s3_geotiff_rdd
+        Reading a GeoTiff from S3 that does have a time component. An option
+        parameter is also passed in which specifies that each GeoTiff should
+        be cut up into 256x256 tiles.
 
-    def get_rdd(self, key_type, bucket, prefix, options=None):
-        key = self.geopysc.map_key_input(key_type, False)
+        >>> geotiff_rdd(geopysc, SPACETIME, 's3://bucket/data/september_images/', maxTileSize=256)
+        RDD
+    """
 
-        if options is None:
-            result = self._s3_wrapper.getRDD(key, bucket, prefix, self.geopysc.sc)
+    key = geopysc.map_key_input(rdd_type, False)
+
+    if kwargs and not options:
+        options = kwargs
+
+    if 's3' in uri:
```

Comment:
geotrellis uses an Rd for this, should at least lift that if there isn't standard uri object for python